localstack-core/localstack/services/lambda_/invocation/assignment.py

```diff
@@ -75,7 +76,10 @@ def get_environment(
 
         try:
             yield execution_environment
-            execution_environment.release()
+            if is_lambda_debug_timeout_enabled_for(lambda_arn=function_version.qualified_arn):
+                self.stop_environment(execution_environment)
```

Comment:
Is this something the users expect? This would effectively stop the possibility of leaving the debugger attached for multiple invocations, which should be theoretically possible currently, right?

Does it make sense, maybe, to only stop the environment if something actually changed for it?

Reply:
Leaving the debugger attached is definitely the behaviour we envision. However, this is not an experience we can offer consistently at the moment. I cannot get this to work for python for example. Currently it is impossible to debug a lambda function more than once.
I am of the idea of providing this behaviour to unblock, whilst we continue the work on integrating more tightly the Lambda Debug Mode and the debuggers. What do you think?

Reply:
That is fine by me, let's keep it as a temporary workaround!